src/datachain/model/pose.py

```diff
@@ -19,22 +24,76 @@ class Pose(DataModel):
     y: list[int] = Field(default=[])
 
     @staticmethod
-    def from_list(points: list[list[float]]) -> "Pose":
+    def from_list(
+        points: Sequence[Sequence[float]],
+        normalized_to: Optional[Sequence[int]] = None,
+    ) -> "Pose":
+        """
+        Create a Pose instance from a list of x and y coordinates.
+
+        If the input coordinates are normalized (i.e., floats between 0 and 1),
+        they will be converted to absolute pixel values based on the provided
+        image size. The image size should be given as a tuple (width, height)
+        via the `normalized_to` argument.
+
+        Args:
+            points (Sequence[Sequence[float]]): The x and y coordinates
+                of the keypoints. List of 2 lists: x and y coordinates.
+            normalized_to (Sequence[int], optional): The reference image size
+                (width, height) for denormalizing the bounding box. If None (default),
+                the coordinates are assumed to be absolute pixel values.
+
+        Returns:
+            Pose: A Pose object.
+        """
+        assert isinstance(points, (tuple, list)), "Pose must be a list of 2 lists."
         assert len(points) == 2, "Pose must be a list of 2 lists: x and y coordinates."
         points_x, points_y = points
+        assert isinstance(points_x, (tuple, list)) and isinstance(
+            points_y, (tuple, list)
+        ), "Pose must be a list of 2 lists."
         assert len(points_x) == len(points_y) == 17, (
             "Pose x and y coordinates must have the same length of 17."
         )
         assert all(
             isinstance(value, (int, float)) for value in [*points_x, *points_y]
         ), "Pose coordinates must be floats or integers."
+
+        if normalized_to is not None:
+            assert all(0 <= coord <= 1 for coord in [*points_x, *points_y]), (
+                "Normalized coordinates must be floats between 0 and 1."
+            )
+            width, height = validate_img_size(normalized_to)
+            points_x = [coord * width for coord in points_x]
```

Comment:
can it be a vectorized operation with numpy?

Reply:
Just wonder if it will be faster for the list of 17 items 🤔